internal/printer/printer.go

```diff
@@ -341,17 +340,45 @@ func (w *writer) printDict(d *ast.Dict, indent int32) {
 	w.print("}")
 }
 
-func (w *writer) printFor(n *ast.For, indent int32) {
-	w.print("for ")
-	w.printNode(n.Target, indent)
-	w.print(" in ")
-	w.printNode(n.Iter, indent)
-	w.print(":\n")
-	for i, node := range n.Body {
+func (w *writer) printFor(n *ast.For, isAsync bool, indent int32) {
+	// We should always have a body
+	if len(n.Body) <= 0 {
+		panic(n)
+	}
+
+	// TODO: How to better support list comprehension? Maybe add a flag to AST node ForNode and AsyncNode?
+	_, isCall := n.Body[0].Node.(*ast.Node_Call)
+	if len(n.Body) == 1 && isCall {
+		w.print("[\n")
 		w.printIndent(indent + 1)
-		w.printNode(node, indent+1)
-		if i != len(n.Body)-1 {
-			w.print("\n")
+		w.printNode(n.Body[0], indent+1)
+		w.print("\n")
+		w.printIndent(indent + 1)
+		if isAsync {
+			w.print("async ")
+		}
+		w.print("for ")
+		w.printNode(n.Target, indent)
+		w.print(" in ")
+		w.printNode(n.Iter, indent)
+		w.print("\n")
+		w.printIndent(indent)
+		w.print("]\n")
+	} else {
+		if isAsync {
+			w.print("async ")
+		}
+		w.print("for ")
+		w.printNode(n.Target, indent)
+		w.print(" in ")
+		w.printNode(n.Iter, indent)
+		w.print(":\n")
+		for i, node := range n.Body {
+			w.printIndent(indent + 1)
+			w.printNode(node, indent+1)
+			if i != len(n.Body)-1 {
+				w.print("\n")
+			}
 		}
 	}
```

Comment:
@kyleconroy I am not so happy with this list comprehension implementation. How would you like to approach this? Add an extra type to the Python AST protobuf or a flag or?

Reply:
@kyleconroy now that the `ast.proto` is available, I can fix this indeed with a flag or a different node. Any preference or leave as is for now?